File: src/main/java/com/dh/ondot/notification/domain/EmergencyAlert.java
package com.dh.ondot.notification.domain;

import com.dh.ondot.core.util.DateTimeUtils;
import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;
import java.time.LocalDateTime;

@Entity
@Table(name = "emergency_alerts")
@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class EmergencyAlert {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "emergency_alert_id")
    Long id;

    @Column(name="content", columnDefinition="TEXT")
    String content;

    @Column(name="region_name")
    String regionName;

    @Column(name="created_at", nullable = false, unique = true)
    Instant createdAt;

    public static EmergencyAlert create(
            String content,
            String regionName,
            LocalDateTime createdAt
    ) {
        return EmergencyAlert.builder()
                .content(content)
                .regionName(regionName)
                .createdAt(DateTimeUtils.toInstant(createdAt))
                .build();
    }
}

File: src/main/kotlin/com/dh/ondot/notification/domain/EmergencyAlert.kt
package com.dh.ondot.notification.domain

import com.dh.ondot.core.util.TimeUtils
import jakarta.persistence.*
import java.time.Instant
import java.time.LocalDateTime

@Entity
@Table(name = "emergency_alerts")
class EmergencyAlert(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "emergency_alert_id")
    val id: Long = 0L,

    @Column(name = "content", columnDefinition = "TEXT")
    val content: String = "",

    @Column(name = "region_name")
    val regionName: String = "",

    @Column(name = "created_at", nullable = false, unique = true)
    val createdAt: Instant = Instant.now(),
) {
    companion object {
        fun create(content: String, regionName: String, createdAt: LocalDateTime): EmergencyAlert =
            EmergencyAlert(
                content = content,
                regionName = regionName,
                createdAt = TimeUtils.toInstant(createdAt),
            )
    }
}
